src/v/pandaproxy/json/requests/fetch.h

```diff
@@ -42,22 +42,48 @@ class rjson_serialize_impl<model::record> {
       , _tpv(tpv)
       , _base_offset(base_offset) {}
 
-    void
+    bool
     operator()(::json::Writer<::json::StringBuffer>& w, model::record record) {
+        auto offset = _base_offset() + record.offset_delta();
+
         w.StartObject();
         w.Key("topic");
         ::json::rjson_serialize(w, _tpv.topic);
         w.Key("key");
-        rjson_serialize_fmt(_fmt)(w, record.release_key());
+        if (!rjson_serialize_fmt(_fmt)(w, record.release_key())) {
+            throw serialize_error(
+              make_error_code(error_code::unable_to_serialize),
+              fmt::format(
+                "Unable to serialize key of record at offset {} in "
+                "topic:partition {}:{}",
+                offset,
+                _tpv.topic(),
+                _tpv.partition()));
+        }
         w.Key("value");
-        rjson_serialize_fmt(_fmt)(w, record.release_value());
+        if (!rjson_serialize_fmt(_fmt)(w, record.release_value())) {
+            throw serialize_error(
+              make_error_code(error_code::unable_to_serialize),
+              fmt::format(
+                "Unable to serialize value of record at offset {} in "
+                "topic:partition {}:{}",
+                offset,
+                _tpv.topic(),
+                _tpv.partition()));
+        }
         w.Key("partition");
         ::json::rjson_serialize(w, _tpv.partition);
         w.Key("offset");
-        ::json::rjson_serialize(w, _base_offset() + record.offset_delta());
+        ::json::rjson_serialize(w, offset);
         w.EndObject();
+
+        return true;
```

Comment:
is there any reason to return bool? it seems like the method either succeeds or throws.

Reply:
It's a little convoluted.

`rjson_serialize_impl<iobuf>::operator()` returned a boolean.  This was called from `rjson_serialize_fmt_impl<T>::operator()`.  the `rjson_serialize_fmt_impl` originally did not return anything.  I updated that to return a `bool` in order to return the result of `rjson_serialize_impl<iobuf>::operator()`.  However that meant I had to change all implementations of `rjson_serialize_impl<T>::operator()` to return a `bool`.